eam-light-backendweb/src/main/java/ch/cern/cmms/eamlightweb/workorders/partusage/PartUsageRest.java

```diff
@@ -96,6 +101,64 @@ public Response loadBinList(@QueryParam("transaction") String transaction, @Quer
 		}
 	}
 
+	@GET
+	@Path("/lots")
+	@Produces("application/json")
+	@Consumes("application/json")
+	public Response loadLotList(@QueryParam("transaction") String transaction, @QueryParam("lot") String lot,
+								@QueryParam("bin") String bin, @QueryParam("part") String part,
+								@QueryParam("store") String store,
+								@QueryParam("requireAvailableQty") boolean requireAvailableQty) {
+		try {
+			GridRequest gridRequest;
+			InforContext context = authenticationTools.getInforContext();
+			String gridFieldLotCode;
+			String gridFieldDescriptionCode = "2175";
+
+			if (transaction.equals("ISSUE")) {
+				gridRequest = new GridRequest("LVIRLOT", GridRequest.GRIDTYPE.LOV);
+				gridFieldLotCode = "825";
+
+				gridRequest.addParam("bin_code", bin);
+				gridRequest.addParam("part_code", part);
+				gridRequest.addParam("part_org", context.getOrganizationCode());
+				gridRequest.addParam("store_code", store);
+
+				if (requireAvailableQty) {
+					gridRequest.addFilter("availableqty", "0", ">", GridRequestFilter.JOINER.AND);
+				}
+
+			} else if (transaction.equals("RETURN")) {
```

Comment:
Having a different method for each transaction case would make the code much more readable

Reply:
I agree, I was basing myself on the method above assuming that was the preferred way. Just let me know, and I'll change it

Reply:
If there is no time constraint we go for the cleaner approach

Reply:
I don't think there is, I'll change it then